sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -93,5 +95,40 @@ private class AssignmentExpressionBannedWordsFinder : CredentialWordsFinderBase<
                 syntaxNode.Left.IsKnownType(KnownType.System_String, semanticModel) &&
                 syntaxNode.Right.IsKind(SyntaxKind.StringLiteralExpression);
         }
+
+        private class StringLiteralBannedWordsFinder : CredentialWordsFinderBase<LiteralExpressionSyntax>
+        {
+            public StringLiteralBannedWordsFinder(DoNotHardcodeCredentialsBase<SyntaxKind> analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(LiteralExpressionSyntax syntaxNode) =>
+                syntaxNode.GetStringValue();
+
+            protected override string GetVariableName(LiteralExpressionSyntax syntaxNode) =>
+                null;
+
+            protected override bool IsAssignedWithStringLiteral(LiteralExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.IsKind(SyntaxKind.StringLiteralExpression) && !IsHandledByOtherFinder(syntaxNode.GetTopMostContainingMethod(), syntaxNode);
+
+            private static bool IsHandledByOtherFinder(SyntaxNode method, SyntaxNode current)
+            {
+                while (current != null && current != method)
+                {
+                    switch (current.Kind())
+                    {
+                        case SyntaxKind.VariableDeclarator:
+                        case SyntaxKind.SimpleAssignmentExpression:
+                            return true;
+                        case SyntaxKind.InvocationExpression:
+                        case SyntaxKind.Argument:
+                        case SyntaxKind.AddExpression: // String concatenation is not supported by other finders
+                            return false;
+                        default:
+                            current = current.Parent;
+                            break;
+                    }
+                }
+                return false;
```

Comment:
we're going to return `false` anyway... I understand the early return above, the problem is that it seems we won't raise issues only for `SyntaxKind.InvocationExpression`, `Argument` and `AddExpression`

do we know, then, all the cases where we will raise issues?

I would make this function differently: `ShouldHandle`
- returns `true` if it's one of the cases we know we want to handle
- `false` otherwise

Reply:
There are many different situations where string can appear in source code. Therefore can appear compiled inside assembly violating the S2068. We don't know list of all possible scenarios.